framework/include/mesh/MooseMesh.h

```diff
@@ -1575,6 +1592,15 @@ class MooseMesh : public MooseObject, public Restartable, public PerfGraphInterf
                             int child,
                             int child_side);
 
+  void buildHRefinementAndCoarseningMaps(Assembly * assembly);
```

Comment:
I ll look at the implementation. Not sure whether using assembly directly in the mesh is where we want to be.
Maybe a util is warranted